### ISSUE TITLE
Netty Handler-proxy could use a factory

### ISSUE THREAD

**Issue by CodingFabian** [netty/netty#4125]
Dear Netty people,

adding Proxy support to a netty client is really easy. However I needed this piece of code. Do you think that this is common and should be part of netty? Feel free to take it :-)

PS: the use case is clear, right? Many tools will expose proxy settings as configuration, so everybody needs to parse it then somehow

    public class ProxyFactory {
    
      public static ProxyHandler make(String proxyType, String proxyHost, int proxyPort, String proxyUser,
          String proxyPassword) {
        InetSocketAddress proxyAddress = new InetSocketAddress(proxyHost, proxyPort);
        String type = proxyType.trim();
        if ("http".equalsIgnoreCase(type)) {
          if (StringUtils.isBlank(proxyUser)) {
            return new HttpProxyHandler(proxyAddress);
          } else {
            return new HttpProxyHandler(proxyAddress, proxyUser, proxyPassword);
          }
        } else if ("socks4".equalsIgnoreCase(type)) {
          if (StringUtils.isBlank(proxyUser)) {
            return new Socks4ProxyHandler(proxyAddress);
          } else {
            return new Socks4ProxyHandler(proxyAddress, proxyUser);
          }
        } else if ("socks5".equalsIgnoreCase(type)) {
          if (StringUtils.isBlank(proxyUser)) {
            return new Socks5ProxyHandler(proxyAddress);
          } else {
            return new Socks5ProxyHandler(proxyAddress, proxyUser, proxyPassword);
          }
        }
        throw new RuntimeException("Invalid proxy type " + type);
      }
    
    }

**Comment by trustin** [comment]
Yeah, such a factory method would be useful for sure. Care to send a pull request?

**Comment by normanmaurer** [comment]
+1

**Comment by CodingFabian** [comment]
ok, cool I will beautify and document this and send a PR

**Comment by normanmaurer** [comment]
<3
